### PR TITLE
Dev 6

### PR DESCRIPTION
**Pull request by Nokafan**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dto/UserDto.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.dev.cinema.dto;
+
+import lombok.Data;
+
+@Data
+public class UserDto {
+    private Long id;
+    private String email;
+    private String password;
+    private byte[] salt;
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dto/UserDto.java]: Redundant

File: src/main/java/com/dev/cinema/dto/moviesession/MovieSessionResponseDto.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.dev.cinema.model.CinemaHall;
+import com.dev.cinema.model.Movie;
+import java.time.LocalDateTime;
+import lombok.Data;
+
+@Data
+public class MovieSessionResponseDto {
+    private Long id;
+    private Movie movie;
+    private CinemaHall cinemaHall;
```
Comment (dariiabutorina) [inline src/main/java/com/dev/cinema/dto/moviesession/MovieSessionResponseDto.java]: Let's pass id's or title / description

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dto/UserDto.java (hunk 1 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/UserDto.java]: Agree, removed it. Thx.

File: src/main/java/com/dev/cinema/dto/moviesession/MovieSessionResponseDto.java (hunk 2 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/moviesession/MovieSessionResponseDto.java]: Ok. Fixed.

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    @PostMapping("/registration")
+    public void registration(@RequestBody UserDto userDto) {
+        service.register(userDto.getEmail(), userDto.getPassword());
+    }
+
+    @PostMapping("/login")
+    public void login(@RequestBody UserDto userDto) throws AuthenticationException {
+        service.login(userDto.getEmail(), userDto.getPassword());
+    }
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: let's remove this method for now - we will implement it later
and use @RestController annotation

File: src/main/java/com/dev/cinema/service/MovieSessionService.java [hunk 4]
```diff
@@ -7,5 +7,7 @@
 public interface MovieSessionService {
     List<MovieSession> findAvailableSessions(Long movieId, LocalDate date);
 
+    MovieSession get(Long id);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/service/MovieSessionService.java]: it is not necessary to add methods to the service layer that you do not use from the controller layer

File: src/main/java/com/dev/cinema/dto/CinemaHallDto.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.dto;
+
+import lombok.Data;
+
+@Data
+public class CinemaHallDto {
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dto/CinemaHallDto.java]: let's remove this dto - we don't use it at the moment

File: src/main/java/com/dev/cinema/dto/UserDto.java [hunk 6]
```diff
@@ -0,0 +1,10 @@
+package com.dev.cinema.dto;
+
+import lombok.Data;
+
+@Data
+public class UserDto {
+    private Long id;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/dto/UserDto.java]: do we need this field?

File: src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionMapper {
+    private final MovieService movieService;
+    private final CinemaHallService hallService;
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy.MM.dd HH:mm");
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java]: should it be a constant field?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/controllers/AuthenticationController.java (hunk 3 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/controllers/AuthenticationController.java]: Changed to RestController, removed /login

File: src/main/java/com/dev/cinema/service/MovieSessionService.java (hunk 4 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/service/MovieSessionService.java]: Ok. np.

File: src/main/java/com/dev/cinema/dto/CinemaHallDto.java (hunk 5 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/CinemaHallDto.java]: Ok

File: src/main/java/com/dev/cinema/dto/UserDto.java (hunk 6 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/UserDto.java]: We have the method get by id so we could use this field for that case. We could receive requests get by id for example.

File: src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java (hunk 7 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java]: I am not sure. I did it for the purpose to be sure that all saved dates and times will be in the same format. It is possible to make a mix of different types of format if not use it. (2020.07.07 or 2020-07-07 etc.) Do I have to remove it?

Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java]: just make it a constant field :)

File: src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MovieSessionMapper {
+    private final MovieService movieService;
+    private final CinemaHallService hallService;
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy.MM.dd HH:mm");
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java]: let's make it a constant field

File: src/main/java/com/dev/cinema/dto/UserDto.java (hunk 6 above)
Comment (KseniiaMakarova) [reply src/main/java/com/dev/cinema/dto/UserDto.java]: let's remove this field

if we would like to find user by id, we would not use DTO
we would rather pass the id as @PathVariable or @RequestParam

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dto/UserDto.java (hunk 6 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/dto/UserDto.java:7]: Removed. Could I use this UserDto as RequestUserDto and add one more ResponceUserDto with id as well and send this response with id in the future hw?

File: src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java (hunk 8 above)
Comment (Nokafan) [reply src/main/java/com/dev/cinema/mapper/MovieSessionMapper.java]: Fixed to constant.

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**